src/main/java/school/hei/haapi/endpoint/rest/mapper/GradeMapper.java

```diff
@@ -0,0 +1,56 @@
+package school.hei.haapi.endpoint.rest.mapper;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.endpoint.rest.model.ExamDetail;
+import school.hei.haapi.endpoint.rest.model.StudentExamGrade;
+import school.hei.haapi.endpoint.rest.model.StudentGrade;
+import school.hei.haapi.model.Exam;
+import school.hei.haapi.endpoint.rest.model.Grade;
+import school.hei.haapi.model.User;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@AllArgsConstructor
+public class GradeMapper {
+  public school.hei.haapi.model.Grade toDomain(Grade grade) {
+    return school.hei.haapi.model.Grade.builder().score(grade.getScore().intValue())
+        .creationDatetime(grade.getCreatedAt()).build();
+  }
+
+  public Grade toRest(school.hei.haapi.model.Grade grade) {
+    return new Grade().id(grade.getId()).createdAt(grade.getCreationDatetime())
+        .score(Double.valueOf(grade.getScore()));
+  }
+
+  public StudentGrade toRestStudentGrade(school.hei.haapi.model.Grade grade) {
+    if (grade == null) {
+      return null;
+    }
+    return new StudentGrade().id(grade.getStudent().getId()).ref(grade.getStudent().getRef())
+        .firstName(grade.getStudent().getFirstName()).lastName(grade.getStudent().getLastName())
+        .email(grade.getStudent().getEmail()).grade(toRest(grade));
+  }
+
+  public StudentExamGrade toRestStudentExamGradeFromStudentAndExam(User student, Exam exam) {
```

Comment:
Just toRest. 

Surcharge method instead of doing this. If necessary, split it inside the appropriate Mapper component.